File: Patterns/10. Hourglass Patterns/93. Hourglass Increasing Numbers/hin.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main(){
    int n,row;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=2*n-1; i++){

        row = min(i,2*n-i);

        // print spaces
        for(int j=1; j<=row-1; j++){
            cout<<"  ";
        }
        // print first half
        for(int j=1; j<=n-row+1; j++){
            cout<<j<<" ";
        }
        // print second half
        for(int j=n-row; j>=1; j--){
            cout<<j<<" ";
        }
        cout<<'\n';

    }
}

/*
Output

1 2 3 4 5 4 3 2 1 
  1 2 3 4 3 2 1 
    1 2 3 2 1 
      1 2 1 
        1 
      1 2 1 
    1 2 3 2 1 
  1 2 3 4 3 2 1 
1 2 3 4 5 4 3 2 1 

*/

File: Patterns/10. Hourglass Patterns/93. Hourglass Increasing Numbers/hin.py
n = int(input("Enter a Number : "))

for i in range(1,2*n):

    row = min(i,2*n-i)
    
    # print spaces
    for j in range(1,row):
        print(end='  ')
    # print first half
    for j in range(1,n-row+2):
        print(j, end=' ')
    # print second half
    for j in range(n-row,0,-1):
        print(j, end=' ')
    print()


'''
Output:

1 2 3 4 5 4 3 2 1 
  1 2 3 4 3 2 1 
    1 2 3 2 1 
      1 2 1 
        1 
      1 2 1 
    1 2 3 2 1 
  1 2 3 4 3 2 1 
1 2 3 4 5 4 3 2 1  


'''